lib/qesapdeployment.pm

```diff
@@ -745,4 +749,112 @@ sub qesap_get_az_resource_group {
+
+=back
+
+=cut
+
+sub qesap_calculate_az_address_range {
+    my %args = @_;
+    croak 'Missing mandatory slot argument' unless $args{slot};
+
+    die "Invalid 'slot' argument - valid values are 1-8192" if ($args{slot} > 8192 || $args{slot} < 1);
+
+    my $offset = ($args{slot} - 1) * 8;
+
+    # addresses are of the form 10.ip2.ip3.0/21 and /24 respectively
+    #ip2 gets incremented when is >=256
+    my $ip2 = int($offset / 256);
+    #ip3 gets incremented by 8 until it's >=256, then it resets
+    my $ip3 = $offset % 256;
+
+    my $vnet_address_range = sprintf("10.%d.%d.0/21", $ip2, $ip3);
+    my $subnet_address_range = sprintf("10.%d.%d.0/24", $ip2, $ip3);
+
+    return (
+        vnet_address_range => $vnet_address_range,
+        subnet_address_range => $subnet_address_range,
+    );
+}
+
+=head3 qesap_az_vnet_peering
+
+    Performs peering between the cluster and IBS mirror on Azure.
+
+=over 2
+
+=item B<SOURCE_GROUP> - resource group of source
+
+=item B<TARGET_GROUP> - resource group of target
+
+=back
+=cut
+
+sub qesap_az_vnet_peering {
+    my (%args) = @_;
+    croak 'Missing mandatory source_group argument' unless $args{source_group};
+    croak 'Missing mandatory target_group argument' unless $args{target_group};
+    $args{source_vnet} = script_output('az network vnet list --resource-group ' . $args{source_group} . ' --query "[0].name" -o tsv');
+    $args{target_vnet} = script_output('az network vnet list --resource-group ' . $args{target_group} . ' --query "[0].name" -o tsv');
+
+    my $source_vnet_id = script_output("az network vnet show --resource-group $args{source_group} --name $args{source_vnet} --query id --output tsv");
+    record_info("[M] source vnet ID: $source_vnet_id\n");
+
+    my $target_vnet_id = script_output("az network vnet show --resource-group $args{target_group} --name $args{target_vnet} --query id --output tsv");
+    record_info("[M] target vnet ID: $target_vnet_id\n");
+
+    my $peering_name = "$args{source_vnet}-$args{target_vnet}";
+
+    assert_script_run("az network vnet peering create --name $peering_name --resource-group $args{source_group} --vnet-name $args{source_vnet} --remote-vnet $target_vnet_id --allow-vnet-access --output table");
+    record_info("PEERING SUCCESS (source)", "[M] Peering from $args{source_group}.$args{source_vnet} server was successful\n") if $args{verbose};
+
+    assert_script_run("az network vnet peering create --name $peering_name --resource-group $args{target_group} --vnet-name $args{target_vnet} --remote-vnet $source_vnet_id --allow-vnet-access --output table");
+    record_info("PEERING SUCCESS (target)", "[M] Peering from $args{target_group}.$args{target_vnet} server was successful\n") if $args{verbose};
+
+    enter_cmd("echo \"[M] Peering status:\n\"");
+    assert_script_run("az network vnet peering show --name $peering_name --resource-group $args{target_group} --vnet-name $args{target_vnet} --output table");
```

Comment:
I'm guessing these 2 lines above will leave something like this in the serial terminal:
```
install# echo "[M] Peering status:
"
[M] Peering status:

install# az network vnet peering show --name <PEERING NAME> --resource-group <RES GRP> --vnet-name <TGT VNET> --output table
```
And then the output of the az-cli command.

Perhaps it would look better from a review standpoint instead to do:
```
record_info '[M] Peering Status',
    script_output("az network vnet peering show --name $peering_name --resource-group $args{target_group} --vnet-name $args{target_vnet} --output table");
```
Or something similar.

Reply:
True. This is mostly me trying to imitate the original bash script by Jan exactly. Will implement your solution.